http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -861,60 +885,34 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       }
     } yield ()
 
-  private def deleteDiskResource(id: Either[Long,
-                                            WsmControlledResourceId
-                                 ], // can be either runtimeId or diskResourceId in order to get WSMcontrolled resource
-                                 workspaceId: WorkspaceId,
-                                 auth: Authorization
-  )(implicit
-    ev: Ask[F, AppContext]
-  ): F[Unit] =
+  private def deleteDiskResource(wsmResourceId: WsmControlledResourceId, workspaceId: WorkspaceId, auth: Authorization)(
+    implicit ev: Ask[F, AppContext]
+  ): F[WsmJobId] =
     for {
       ctx <- ev.ask
-      diskResourceOpt <- id match {
-        case Left(runtimeId) =>
-          controlledResourceQuery
-            .getWsmRecordForRuntime(runtimeId, WsmResourceType.AzureDisk)
-            .transaction
-        case Right(diskResourceId) =>
-          controlledResourceQuery
-            .getWsmRecordFromResourceId(diskResourceId, WsmResourceType.AzureDisk)
-            .transaction
-      }
-      diskResource <- F.fromOption(
-        diskResourceOpt,
-        AzureDiskResourceDeletionError(id,
-                                       workspaceId,
-                                       "No disk resource found for delete azure disk. No-op for wsmDao.deleteDisk."
-        )
-      )
-      jobId = getWsmJobId("delete-disk", diskResource.resourceId)
-
-      _ <- diskResourceOpt.traverse { disk =>
-        for {
-          _ <- logger.info(ctx.loggingCtx)(s"Sending WSM delete message for disk resource ${disk.resourceId.value}")
-          _ <- wsmDao
-            .deleteDisk(
-              DeleteWsmResourceRequest(
-                workspaceId,
-                disk.resourceId,
-                DeleteControlledAzureResourceRequest(
-                  WsmJobControl(jobId)
-                )
-              ),
-              auth
-            )
-            .void
-            .adaptError(e =>
-              AzureDiskDeletionError(
-                disk.resourceId,
-                workspaceId,
-                s"${ctx.traceId.asString} | WSM call to delete disk failed due to ${e.getMessage}. Please retry delete again"
-              )
+      jobId = getWsmJobId("delete-disk", wsmResourceId)
+
+      _ <- logger.info(ctx.loggingCtx)(s"Sending WSM delete message for disk resource ${wsmResourceId.value}")
+      _ <- wsmDao
+        .deleteDisk(
+          DeleteWsmResourceRequest(
+            workspaceId,
+            wsmResourceId,
+            DeleteControlledAzureResourceRequest(
+              WsmJobControl(jobId)
             )
-        } yield ()
-      }.void
-    } yield ()
+          ),
+          auth
+        )
+        .void
+        .adaptError(e =>
+          AzureDiskDeletionError(
+            wsmResourceId,
+            workspaceId,
+            s"${ctx.traceId.asString} | WSM call to delete disk failed due to ${e.getMessage}. Please retry delete again"
```

Comment:
it looks like here the errors being thrown (if I am following this correctly) are raising an error, _but_ the ask is that we retry the delete (even though the resources don't exist). am I missing something else going on here? is this exception's usage being overloaded?

Reply:
> but the ask is that we retry the delete (even though the resources don't exist)
I'm not sure I fully understand here, this is when the deletion errors in WSM for some reason and it happens very rarely, the only time I've seen it recently was a `Received duplicate jobId` error.

Reply:
sorry ... I had to re-read a couple times to make sure I understood, and now I understand the comment better, and your code is fine. sorry to be confusing. 

I thought wrongly that this was erroring if we attempted to delete a disk that doesn't exist, but that is not the case here .